File: webapi/DAL/Entities/Main/Category.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace webapi.DAL.Entities.Main
{
    [Table("categories")]
    public class Category : Entity
    {
        [Column("name")]
        public string Name { get; set; }

        [Column("image")]
        public string Image { get; set; }

        public List<SubCategory> SubCategories { get; set; }
    }
}

File: angularapp/src/app/interfaces/category.ts
export interface Category {
  id: number;
  name: string;
  image: string;
}

export interface CategoryModel1 {
  id: number;
  name: string;
  image: string;
  subCategories: SubCategoryModel1[];
}

export interface SubCategoryModel1 {
  id: number;
  name: string;
}
